glide-core/redis-rs/redis/src/cluster_async/pipeline_routing.rs

```diff
@@ -430,86 +484,766 @@ pub fn add_pipeline_result(
+                redirect: redirect_info,
+                previous_routing: Box::new(InternalSingleNodeRouting::Random::<C>),
+            };
+
+            // Retrieve the original command and attempt to get a new connection.
+            match get_original_cmd(pipeline, index, inner_index, Some(response_policies)) {
+                Ok(cmd) => {
+                    match ClusterConnInner::get_connection(routing, core.clone(), None).await {
+                        Ok((address, conn)) => {
+                            // If the retry method is AskRedirect, add `ASKING` before the actual command.
+                            if matches!(retry_method, RetryMethod::AskRedirect) {
+                                add_asking_command_to_node_pipeline_map(
+                                    pipeline_map,
+                                    address.clone(),
+                                    conn.clone(),
+                                    index,
+                                    inner_index,
+                                );
+                            }
+                            // Add the command to the node pipeline map to retry.
+                            add_command_to_node_pipeline_map(
+                                pipeline_map,
+                                address,
+                                conn,
+                                cmd,
+                                index,
+                                inner_index,
+                            );
+                            continue;
+                        }
+                        Err(err) => error.append_detail(&err.into()), // Failed to get a connection.
+                    }
+                }
+                Err(cmd_err) => error.append_detail(&cmd_err), // Failed to get the original command.
+            }
+        } else {
+            let server_error = ServerError::ExtensionError {
+                code: "RedirectError".to_string(),
+                detail: Some("Failed to find redirect info".to_string()),
+            };
+            // Failed to get the original command.
+            error.append_detail(&server_error);
+        }
+
+        // Append the error if a failure occurred.
+        add_pipeline_result(
+            pipeline_responses,
+            index,
+            inner_index,
+            Value::ServerError(error),
+            address,
+        )?;
+    }
     Ok(())
 }
 
+/// Handles a MOVED redirection error by updating the cluster topology.
+/// If updating the topology fails, the error is appended to the original pipeline response.
+/// Returns `true` if a failure occurred, otherwise `false`.
+async fn handle_moved_redirect<C>(
+    core: Core<C>,
+    redis_error: &RedisError,
+    error: &mut ServerError,
+    index: usize,
+    inner_index: Option<usize>,
+    pipeline_responses: &mut PipelineResponses,
+    address: &str,
+) -> Result<bool, (OperationTarget, RedisError)>
```

Comment:
```suggestion
) -> Result<(), ServerError>
```

change back to 

or maybe result of results